dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1773,4 +1826,668 @@ public void getDirectEpersonGroups() throws Exception {
                             );
 
     }
+
+    @Test
+    public void patchReplacePasswordWithToken() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("Johndoe@fake-email.com")
+                                        .withPassword(password)
+                                        .build();
+
+        String newPassword = "newpassword";
+
+        context.restoreAuthSystemState();
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/password", newPassword);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+        accountService.sendRegistrationInfo(context, ePerson.getEmail());
+        String tokenForEPerson = registrationDataService.findByEmail(context, ePerson.getEmail()).getToken();
+        String token = getAuthToken(admin.getEmail(), password);
+        PasswordHash oldPassword = ePersonService.getPasswordHash(ePerson);
+        // updates password
+        getClient(token).perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON)
+                                     .param("token", tokenForEPerson))
+                        .andExpect(status().isOk());
+
+        PasswordHash newPasswordHash = ePersonService.getPasswordHash(ePerson);
+        assertFalse(oldPassword.equals(newPasswordHash));
+        assertTrue(registrationDataService.findByEmail(context, ePerson.getEmail()) == null);
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.deleteByToken(context, tokenForEPerson);
+        context.restoreAuthSystemState();
```

Comment:
Maybe I'm misunderstanding something, but if the registration data *is null*, then shouldn't the token have been deleted already?  Why do we need to have the `deleteByToken()` here?

Reply:
It's indeed not necessary, it has now been replaced with a verification proving the token has been deleted